ext/cl_khr_extended_versioning.asciidoc

```diff
@@ -0,0 +1,286 @@
+// Copyright 2019 The Khronos Group. This work is licensed under a
+// Creative Commons Attribution 4.0 International License; see
+// http://creativecommons.org/licenses/by/4.0/
+
+[[cl_khr_extended_versioning]]
+== Extended versioning
+
+This extension introduces new platform and device queries that return detailed
+version information to applications. It makes it possible to return the exact
+revision of the specification or intermediate languages supported by an
+implementation. It also enables implementations to communicate a version
+number for each of the extensions they support and remove the requirement
+for applications to process strings to test for the presence of an extension or
+intermediate language or built-in kernel.
+
+=== General information
+
+==== Name Strings
+
+`cl_khr_extended_versioning`
+
+==== Contributors
+
+Kévin Petit, Arm Ltd. +
+Ben Ashbaugh, Intel +
+Alastair Murray, Codeplay Software Ltd. +
+Einar Hov, Arm Ltd.
+
+==== Version history
+
+[cols="1,1,3",options="header",]
+|====
+| *Date*     | *Version* | *Description*
+| 2020-02-12 | 1.0.0     | Initial version.
+|====
+
+==== Dependencies
+
+This extension is written against the OpenCL Specification
+Version 2.2, Revision 11.
+
+This extension requires OpenCL 1.0.
+
+=== New API Types
+
+==== Version
+
+This extension introduces a new scheme to encode detailed
+(major, minor, patch/revision) version information into a single 32-bit unsigned
+integer:
+
+* The major version is using bits 31-22
+* The minor version is using bits 21-12
+* The patch version is using bits 11-0
+
+This scheme enables two versions to be ordered using the standard C/C++ operators.
+Macros are provided to extract individual fields or compose a full version
+from the individual fields.
+
+[source,c]
+----
+
+typedef cl_uint cl_version_khr;
+
+#define CL_VERSION_MAJOR_BITS_KHR (10)
+#define CL_VERSION_MINOR_BITS_KHR (10)
+#define CL_VERSION_PATCH_BITS_KHR (12)
+
+#define CL_VERSION_MAJOR_MASK_KHR ((1 << CL_VERSION_MAJOR_BITS_KHR) - 1)
+#define CL_VERSION_MINOR_MASK_KHR ((1 << CL_VERSION_MINOR_BITS_KHR) - 1)
+#define CL_VERSION_PATCH_MASK_KHR ((1 << CL_VERSION_PATCH_BITS_KHR) - 1)
+
+#define CL_VERSION_MAJOR_KHR(version) \
+        ((version) >> (CL_VERSION_MINOR_BITS_KHR + CL_VERSION_PATCH_BITS_KHR))
+#define CL_VERSION_MINOR_KHR(version) \
+        (((version) >> CL_VERSION_PATCH_BITS_KHR) & CL_VERSION_MINOR_MASK_KHR)
+#define CL_VERSION_PATCH_KHR(version) ((version) & CL_VERSION_PATCH_MASK_KHR)
+
+#define CL_MAKE_VERSION_KHR(major, minor, patch) \
+    ((((major) & CL_VERSION_MAJOR_MASK_KHR) << (CL_VERSION_MINOR_BITS_KHR + CL_VERSION_PATCH_BITS_KHR)) | \
+     (((minor) & CL_VERSION_MINOR_MASK_KHR) << CL_VERSION_PATCH_BITS_KHR) | \
+     ((patch) & CL_VERSION_PATCH_MASK_KHR))
+----
+
+==== Name and version
+
+This extension adds a structure that can be used to describe a combination of a
+name alongside a version number:
+
+[source,c]
+----
+#define CL_NAME_VERSION_MAX_NAME_SIZE_KHR 64
```

Comment:
For the main specification this value is in the XML as a constant, this could be too.

Reply:
Done